File: ruby/ruby-sample/basic/variable_scope.rb
# frozen_string_literal: true
# typed: true

require('English')

# variable scope

# global variable
$globalv = 100

# local variable
# starts with lower case letter or underscore
localv = 50

# check scope of variable
puts(defined?($globalv))
puts(defined?(localv))

# system global variables
# see also: https://www.techotopia.com/index.php/Ruby_Variable_Scope

# Ruby Interpreter Process ID
puts $$
# This is equivalent to `$$`
puts $PROCESS_ID

File: php/php-sample/basic/variable_scope.php
<?php

# global variable
$a = 300;
function show_global_variable_scope(): void
{
  # local variable
  $a = 30;
  var_dump($a);

  global $a;
  var_dump($a);
}

show_global_variable_scope();

function show_local_variable_scope(): int
{
  $b = 0;
  $b += 1;
  return $b;
}

var_dump(show_local_variable_scope());
var_dump(show_local_variable_scope());

function show_static_variable_scope(): int
{
  static $b = 0;
  $b += 1;
  return $b;
}

var_dump(show_static_variable_scope());
var_dump(show_static_variable_scope());
